bitwarden/authenticator-ios
# [BITAU-132] Move to Bitwarden Long Press on Local Items

## 🎟️ Tracking

[BITAU-132](https://livefront.atlassian.net/browse/BITAU-132)

## 📔 Objective

This PR adds all of the plumbing needed to store a temporary item and link to the PM app to fetch that temporary item. It is a companion PR to [PR #160](https://github.com/bitwarden/ios/pull/1060) in the PM repo.

In addition, it also adds a new item to the long press menu for local items - Move to Bitwarden. Hitting that button on the long press menu will save the pressed item into the temporary store and launch the PM app with the new Item link.

All of this requires the feature flag to be on and for the user to have already turned on sync for at least one account in the PM app. If they haven't turned on sync yet, we don't show the new long press item.

### Testing Notes
There's a few requirements to properly testing:
- Make sure you've installed the latest from the `main` PM branch and are running that version when testing the deeplink from Authenticator to PM app.
- Make sure the feature flag is turned on in **both** apps.
- Make sure you have already turned on sync in the PM app before testing out the new long press.
- Finally, make sure your are using an `otpauth://` style key since the others are not yet supported.

## 📸 Screenshots
![Simulator Screenshot - iPhone 15 Pro - 2024-10-22 at 12 51 10](https://github.com/user-attachments/assets/8db44d97-c022-4f6b-bd7c-19e238e66599)

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines



- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes
